packages/dev/core/src/Misc/brdfTextureTools.ts

```diff
@@ -50,14 +49,18 @@ export const GetEnvironmentBRDFTexture = (scene: Scene): BaseTexture => {
 
         const observer = scene.getEngine().onContextRestoredObservable.add(() => {
             texture.isRGBD = true;
-            const checkReady = () => {
+            /**
+             * Using scene.onBeforeRenderObservable instead of Tools.SetImmediate to check the texture's state of readiness allows us to check before any rendering occurs.
+             * When a context restore occurs, it gives ExpandRGBDTexture the ability to reset the state to false, preventing the texture from being used in any rendering.
+             * In WebGPU, not doing so would generate an error because ExpandRGBDTexture performs a _swapAndDie on the texture, which causes WebGPU caches to fail if the texture has already been used for rendering.
+             * Only when ExpandRGBDTexture has finished its work, the texture is ready to be used again.
+             */
+            const oo = scene.onBeforeRenderObservable.add(() => {
                 if (texture.isReady()) {

```

Comment:
This is now coupled to rendering whereas before readyness was out of rendering. I am wondering if this could be seen as a breaking change from the users ?

Reply:
I don't think it's a problem because this code only executes after a context restoration. The normal path has not been changed, the BRDF creation is still not tied to scene rendering.